go/teams/loader.go

```diff
@@ -573,20 +575,39 @@ func (l *TeamLoader) load2InnerLockedRetry(ctx context.Context, arg load2ArgT) (
 	var lastLinkID keybase1.LinkID
 	var hiddenIsFresh bool
 
+	// hiddenResp will be nill iff we do not make the merkleLookupWithHidden
+	// call. If the server does not return any hidden data, we will encode that
+	// as a non nil response whose RespType is MerkleHiddenResponseTypeNONE.
+	var hiddenResp *libkb.MerkleHiddenResponse
+
 	if (ret == nil) || repoll {
 		mctx.Debug("TeamLoader looking up merkle leaf (force:%v)", arg.forceRepoll)
-		// Request also, without an additional RTT, freshness information about the hidden chain;
-		// we're going to send up information we know about the visible and hidden chains to both prove
-		// membership and show what we know about.
-		harg, err := hiddenPackage.MerkleLoadArg(mctx)
-		if err != nil {
-			return nil, err
-		}
+
 		// Reference the merkle tree to fetch the sigchain tail leaf for the team.
-		lastSeqno, lastLinkID, hiddenIsFresh, err = l.world.merkleLookupWithHidden(ctx, arg.teamID, arg.public, harg)
+		lastSeqno, lastLinkID, hiddenResp, err = l.world.merkleLookupWithHidden(ctx, arg.teamID, arg.public)
 		if err != nil {
 			return nil, err
 		}
+
+		if hiddenResp.CommittedHiddenTail != nil {
```

Comment:
also factor this into a new subfunction to keep `load2InnerLockedRetry` as unbloated as possible